dotnet/src/Extensions/Reliability.Polly/KernelConfigExtensions.cs

```diff
@@ -0,0 +1,24 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+#pragma warning disable IDE0130 // Namespace does not match folder structure
+
+namespace Microsoft.SemanticKernel.Reliability.Polly;
+
+/// <summary>
+/// Extension methods for <see cref="KernelConfig"/> class.
+/// </summary>
+public static class KernelConfigExtensions
+{
+    /// <summary>
+    /// Sets the default retry configuration for any kernel http request.
+    /// </summary>
+    /// <param name="kernelConfig">Target instance</param>
+    /// <param name="retryConfig">Retry configuration</param>
+    /// <returns>Returns target instance for fluent compatibility</returns>
+    public static KernelConfig SetHttpRetryConfig(this KernelConfig kernelConfig, HttpRetryConfig? retryConfig = null)
```

Comment:
I think we should only support HttpRetryConfig in our own non-polly extension library. Reliability.Polly should be helper utilities for taking a Polly policy and supplying it as part of an IDelegatingHandlerFactory